content/concepts/did-ddo.md

```diff
@@ -1,173 +1,516 @@
 
-DIDs and DDOs follow [this specification](https://w3c-ccg.github.io/did-spec/) defined by the World Wide Web Consurtium (W3C).
+DIDs and DDOs follow the [specification defined by the World Wide Web Consortium (W3C)](https://w3c-ccg.github.io/did-spec/).
 
-## Rules for DIDs & DDOs in Ocean
+## Rules for DIDs & DDOs
 
-- An _asset_ in Ocean represents a downloadable file, compute service, or similar. Each asset is a _resource_ under control of a _publisher_. The Ocean network itself does _not_ store the actual resource (e.g. files).
-- An asset should have a DID and DDO. The DDO should include metadata about the asset.
-- The DDO can only can be modified by _owners_ or _delegated users_.
-- There _must_ be at least one client library acting as _resolver_, to get a DDO from a DID.
-- The DDO is stored on-chain. It's stored in in plaintext, with two exceptions: (1) the field for resource-access url is encrypted (2) the whole DDO may be encrypted, if the publisher is willing to lose 100% of discoverability.
-- A metadata cache like Aquarius can help in reading and writing DDO data from the chain.
+An _asset_ in Ocean represents a downloadable file, compute service, or similar. Each asset is a _resource_ under control of a _publisher_. The Ocean network itself does _not_ store the actual resource (e.g. files).
 
-## DID Structure
+An _asset_ should have a DID and DDO. The DDO should include [metadata](#metadata) about the asset, and define access in at least one [service](#services). The DDO can only can be modified by _owners_ or _delegated users_.
 
-In Ocean, a DID is a string that looks like:
+There _must_ be at least one client library acting as _resolver_, to get a DDO from a DID. A metadata cache like Aquarius can help in reading and searching through DDO data from the chain.
 
-```text
-did:op:0ebed8226ada17fde24b6bf2b95d27f8f05fcce09139ff5cec31f6d81a7cd2ea
-```
+## State
 
-It follows [the generic DID scheme](https://w3c-ccg.github.io/did-spec/#the-generic-did-scheme).
+Each asset has a state, which is held by the NFT contract. The possible states are:
 
-The part after `did:op:` is the asset's on-chain Ethereum address (minus the "0x"). One can be computed from the other; therefore there is a 1:1 mapping between did and Ethereum address.
+- `0` = active
+- `1` = end-of-life
+- `2` = deprecated (by another asset)
+- `3` = revoked by publisher
 
-## DDO Attributes
+## Publishing an Retrieving DDOs
 
-![DDO Content](images/ddo-content.png)
+The DDO is stored on-chain as part of the NFT contract and it is stored encrypted using the private key of the _Provider_. To resolve it, a metadata cache like _Aquarius_ must query the provider to decrypt the DDO.
 
-A DDO has these standard attributes:
+Here is the complete flow:
 
-- `@context`
-- `id`
-- `created`
-- `updated`
-- `publicKey`
-- `authentication`
-- `proof`
-- `verifiableCredential`
+```text
+title DDO flow
```

Comment:
This is hard to read and I know this is some syntax for generating the below image. Can we add a link to the tool which was used here please? So future authors can easily change this

Reply:
Pretty standard UML Sequence Diagram, maybe it's  a bit difficult if you're not familiar with them. Regarding the text before the diagram, not sure what it's purpose is in this context.

Reply:
got it, but yes confusing as now I added something to the text but no image was updated creating mismatch between text & image. If we remove the text part we need a place to store the text as otherwise nobody can update the image.

Gatbsy solution would be https://github.com/baerrach/gatsby-remark-plantuml so we could only keep the text in the source files.

Reply:
got it, but yes confusing as now I added something to the text but no image was updated creating mismatch between text & image. If we remove the text part we need a place to store the text as otherwise nobody can update the image.

Gatbsy solution would be https://github.com/baerrach/gatsby-remark-plantuml so we could only keep the text in the source files.